### ISSUE TITLE
"main.jsbundle does not exist" error when building iOS version locally with Expo SDK 47

### ISSUE THREAD

**Issue by lindan4** [expo/expo#20001]
### Summary

I am trying to build the iOS version of an application locally whose Expo version I had upgraded to version 47. I have no issues building the Android version but run into issues when building the iOS. I can build the iOS version when I generate the Xcode project myself but cannot via eas.

This has not been an issue with SDK 45 or SDK 46.

### What platform(s) does this occur on?

iOS

### Environment

System:
OS: macOS 13.0.1
Shell: 5.8.1 - /bin/zsh
Binaries:
Node: 16.18.1 - /usr/local/bin/node
Yarn: 1.22.19 - /opt/homebrew/bin/yarn
npm: 8.19.2 - /usr/local/bin/npm
Managers:
CocoaPods: 1.11.3 - /Users/lindanthillanayagam/.rvm/gems/ruby-3.0.0/bin/pod
SDKs:
iOS SDK:
Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
IDEs:
Android Studio: 2021.2 AI-212.5712.43.2112.8609683
Xcode: 14.1/14B47b - /usr/bin/xcodebuild
npmPackages:
@expo/webpack-config: ^0.17.2 => 0.17.3
expo: ^47.0.0 => 47.0.3
react: 18.1.0 => 18.1.0
react-dom: 18.1.0 => 18.1.0
react-native: 0.70.5 => 0.70.5
react-native-web: ~0.18.7 => 0.18.10
npmGlobalPackages:
eas-cli: 2.6.0
Expo Workflow: managed

### Minimal reproducible example

1. Initialize an expo project with version 47.
2. Install the packages below:
```
"dependencies": {
    "@expo/webpack-config": "^0.17.2",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-native-community/datetimepicker": "6.5.2",
    "@react-native-community/netinfo": "9.3.5",
    "@react-native-community/slider": "4.2.4",
    "@react-navigation/bottom-tabs": "^6.3.3",
    "@react-navigation/drawer": "^6.4.4",
    "@react-navigation/native": "^6.0.12",
    "@react-navigation/native-stack": "^6.8.0",
    "@sentry/react-native": "4.2.2",
    "@tanstack/query-async-storage-persister": "^4.2.3",
    "@tanstack/react-query": "^4.2.3",
    "@tanstack/react-query-persist-client": "^4.2.1",
    "axios": "^1.1.3",
    "dayjs": "^1.11.5",
    "dotenv": "^16.0.2",
    "expo": "^47.0.0",
    "expo-application": "~5.0.1",
    "expo-auth-session": "~3.7.2",
    "expo-constants": "~14.0.2",
    "expo-device": "~5.0.0",
    "expo-font": "~11.0.1",
    "expo-linking": "~3.2.3",
    "expo-localization": "~14.0.0",
    "expo-location": "~15.0.1",
    "expo-notifications": "~0.17.0",
    "expo-random": "~13.0.0",
    "expo-secure-store": "~12.0.0",
    "expo-splash-screen": "~0.17.4",
    "expo-status-bar": "~1.4.2",
    "expo-updates": "~0.15.4",
    "expo-web-browser": "~12.0.0",
    "i18next": "^21.9.1",
    "i18next-pseudo": "^2.2.1",
    "intl-pluralrules": "^1.3.1",
    "jest": "^26.6.3",
    "jest-expo": "^47.0.0",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-i18next": "^11.18.5",
    "react-native": "0.70.5",
    "react-native-gesture-handler": "~2.8.0",
    "react-native-maps": "1.3.2",
    "react-native-modal": "^13.0.1",
    "react-native-reanimated": "~2.12.0",
    "react-native-safe-area-context": "4.4.1",
    "react-native-screens": "~3.18.0",
    "react-native-svg": "13.4.0",
    "react-native-web": "~0.18.7",
    "sentry-expo": "~5.0.0",
    "styled-components": "^5.3.5"
  },
  "devDependencies": {
    "@babel/core": "^7.19.3",
    "@react-native-community/eslint-config": "^3.1.0",
    "@testing-library/jest-native": "^5.1.2",
    "@testing-library/react-native": "^11.4.0",
    "@types/jest": "^26.0.14",
    "@types/react": "~18.0.24",
    "@types/react-native": "~0.70.6",
    "@types/styled-components": "^5.1.26",
    "@types/styled-components-react-native": "^5.1.3",
    "@typescript-eslint/eslint-plugin": "^5.36.2",
    "@typescript-eslint/parser": "^5.36.2",
    "babel-plugin-react-native-web": "^0.18.9",
    "eas-cli": "^2.6.0",
    "eslint": "^8.23.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-array-func": "^3.1.7",
    "eslint-plugin-import": "^2.25.3",
    "eslint-plugin-jsx-a11y": "^6.6.1",
    "eslint-plugin-promise": "^6.0.1",
    "eslint-plugin-sonarjs": "^0.15.0",
    "license-checker": "^25.0.1",
    "msw": "^0.47.0",
    "prettier": "^2.7.1",
    "react-native-svg-transformer": "^1.0.0",
    "react-test-renderer": "18.1.0",
    "ts-node": "^10.8.1",
    "typescript": "^4.6.3"
  }
```
  
3. Create eas.json file with the following content:

```
{
  "cli": {
    "version": ">= 2.6.0"
  },
  "build": {
    "development": {
      "developmentClient": true,
      "distribution": "internal"
    },
    "preview": {
      "distribution": "internal",
      "ios": {
        "simulator": true
      },
      "android": {
        "buildType": "apk"
      }
    },
  },
  "submit": {
    "production": {}
  }
}
```

4. Run the command `npx eas build --platform ios --non-interactive --local --profile preview`

**Comment by carlnx** [comment]
I am having this issue too :(

**Comment by keith-kurak** [comment]
Thanks for reporting this! We're able to reproduce by creating a blank project and running `eas build -p ios --local`, so it seems like this is an issue and we're looking into it.

**Comment by sebastienlabine** [comment]
@keith-kurak can confirm this is also happening on XCode 13.4.1 (M1) with the SDK 46

**Comment by atanaskanchev** [comment]
Can confirm that the same is happening with my config as well

``` expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.1/bin/yarn
      npm: 9.1.1 - ~/.nvm/versions/node/v16.15.1/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.1/14B47b - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: * => 0.5.1 
      @expo/webpack-config: ^0.17.2 => 0.17.3 
      expo: ^47.0.0 => 47.0.3 
      react: 18.1.0 => 18.1.0 
      react-dom: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
      react-native-web: ~0.18.7 => 0.18.10 
    npmGlobalPackages:
      eas-cli: 2.7.0
      expo-cli: 6.0.8
    Expo Workflow: bare```

**Comment by atanaskanchev** [comment]
Tried downgrading Xcode from 14.1 to 14.0.1, still the same

**Comment by atanaskanchev** [comment]
The remote EAS Build fails as well:
```
Using VM template "macos-monterey-12.4-xcode-13.4" running macOS Monterey 12.4
Installed software:
- Xcode 13.4 (13F17a)
- Node.js 16.13.2
- Yarn 1.22.17
- pnpm 7.0.0
- npm 8.1.2
- fastlane 2.206.1
- CocoaPods 1.11.3
- Ruby 2.7
- Expo CLI 6.0.0

**Comment by eivindml** [comment]
Having the same issue, both for local build and remote EAS build.

**Comment by dsokal** [comment]
@atanaskanchev @eivindml can you share your build IDs?

**Comment by atanaskanchev** [comment]
> @atanaskanchev @eivindml can you share your build IDs?

`54cad59e-9bc9-4902-86d0-2888ee7201fe`

**Comment by eivindml** [comment]
> @atanaskanchev @eivindml can you share your build IDs?

`752577a5-3818-4665-b522-09d24299ab1d`

**Comment by dsokal** [comment]
I looked at your guys' logs. 
@atanaskanchev why do you think it's the **same** issue? I'd say it's not. 
@eivindml it's not an iOS build.

**Comment by eivindml** [comment]
> I looked at your guys' logs.
> @atanaskanchev why do you think it's the **same** issue? I'd say it's not.
> @eivindml it's not an iOS build.

Oh, sorry, I was too quick! 😳

`f81b76c6-b218-49fa-b41c-585ce6743799` Here is the latest iOS build with this failure.

**Comment by atanaskanchev** [comment]
> I looked at your guys' logs.
> @atanaskanchev why do you think it's the **same** issue? I'd say it's not.
> @eivindml it's not an iOS build.

You're right, it has failed on the same spot but for a different reason. Let me try to prebuild it and try again

`
❌  ld: '/Users/expo/Library/Developer/Xcode/DerivedData/LandlordFusion-cezfyocfizloufdazilzsknrvsqo/Build/Intermediates.noindex/ArchiveIntermediates/LandlordFusion/BuildProductsPath/Release-iphoneos/XCFrameworkIntermediates/EXFileSystem/EXFileSystem.framework/EXFileSystem(EXFileSystemAssetLibraryHandler.o)' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. file '/Users/expo/Library/Developer/Xcode/DerivedData/LandlordFusion-cezfyocfizloufdazilzsknrvsqo/Build/Intermediates.noindex/ArchiveIntermediates/LandlordFusion/BuildProductsPath/Release-iphoneos/XCFrameworkIntermediates/EXFileSystem/EXFileSystem.framework/EXFileSystem' for architecture arm64`

**Comment by megan-woodruff** [comment]
I'm seeing the same issue on an EAS ios build, with the same error in XCode logs as @atanaskanchev. 

My build id is `5643e5c8-5004-40f9-bc0d-337004d4893d`.

**Comment by dsokal** [comment]
@eivindml This one is an iOS build indeed but it's not the same issue. Take a look at the Xcode logs phase, download the file, and scroll to the bottom. There's a reason for the failure.
<img width="534" alt="Screenshot 2022-11-14 at 18 45 57" src="https://user-images.githubusercontent.com/5256730/201730027-eb38381b-8a11-4ad9-921b-e6f80f51e1d9.png">

**Comment by eivindml** [comment]
> @eivindml This one is an iOS build indeed but it's not the same issue. Take a look at the Xcode logs phase, download the file, and scroll to the bottom. There's a reason for the failure.
> <img alt="Screenshot 2022-11-14 at 18 45 57" width="534" src="https://user-images.githubusercontent.com/5256730/201730027-eb38381b-8a11-4ad9-921b-e6f80f51e1d9.png">

Great, thank you! I got the same error message from fastlane though, which was a bit confusing.

Also weird that the regular development build didn't complain about the same issue. (Folder name was `app` instead of `App`). But now it works atleast!

**Comment by atanaskanchev** [comment]
An update after cleaning the project and updating to the latest reps.

- Local IOS/Android development builds runs OK
- Local IOS production build fails with `main.jsbundle does not exist`
- EAS remote IOS production build fails with the same issue as https://github.com/expo/expo/issues/20007 (Build ID 605b082a-a2f5-4fd0-b23c-94213f985605)

Using package.json:
```json
{
  "name": "mobile-app",
  "version": "0.0.30",
  "private": true,
  "dependencies": {
    "@babel/core": "^7.19.3",
    "@expo/config-types": "*",
    "@expo/metro-config": "*",
    "@expo/vector-icons": "^13.0.0",
    "@expo/webpack-config": "^0.17.2",
    "@gorhom/bottom-sheet": "*",
    "@hookform/resolvers": "*",
    "@nrwl/expo": "*",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-native-community/datetimepicker": "6.5.2",
    "@react-native-community/netinfo": "9.3.5",
    "@react-native-firebase/app": "*",
    "@react-native-firebase/app-check": "*",
    "@react-native-firebase/auth": "*",
    "@react-native-firebase/crashlytics": "*",
    "@react-native-firebase/firestore": "*",
    "@react-native-firebase/functions": "*",
    "@react-native-firebase/perf": "*",
    "@react-native-firebase/storage": "*",
    "@react-navigation/bottom-tabs": "*",
    "@react-navigation/drawer": "*",
    "@react-navigation/material-top-tabs": "*",
    "@react-navigation/native": "*",
    "@react-navigation/native-stack": "*",
    "@shopify/flash-list": "1.3.1",
    "@tanstack/react-query": "*",
    "@testing-library/jest-native": "*",
    "@testing-library/react-native": "*",
    "clsx": "*",
    "countries-list": "*",
    "currency-codes": "*",
    "currency.js": "*",
    "dayjs": "*",
    "dotenv": "*",
    "expo": "^47.0.0",
    "expo-build-properties": "~0.4.0",
    "expo-constants": "~14.0.2",
    "expo-dev-client": "~2.0.0",
    "expo-font": "~11.0.1",
    "expo-image-picker": "~14.0.1",
    "expo-linking": "~3.2.3",
    "expo-splash-screen": "~0.17.5",
    "expo-status-bar": "~1.4.2",
    "expo-structured-headers": "*",
    "expo-system-ui": "~2.0.1",
    "expo-updates": "~0.15.5",
    "expo-web-browser": "~12.0.0",
    "firebase": "*",
    "intl": "*",
    "luxon": "*",
    "native-base": "*",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-hook-form": "*",
    "react-native": "0.70.5",
    "react-native-chart-kit": "*",
    "react-native-floating-action": "*",
    "react-native-gesture-handler": "~2.8.0",
    "react-native-get-random-values": "~1.8.0",
    "react-native-gifted-chat": "*",
    "react-native-google-places-autocomplete": "*",
    "react-native-keyboard-aware-scroll-view": "*",
    "react-native-modal-datetime-picker": "*",
    "react-native-pager-view": "6.0.1",
    "react-native-reanimated": "~2.12.0",
    "react-native-safe-area-context": "4.4.1",
    "react-native-screens": "~3.18.0",
    "react-native-svg": "13.4.0",
    "react-native-swipe-list-view": "*",
    "react-native-tab-view": "*",
    "react-native-web": "~0.18.7",
    "react-query": "*",
    "rooks": "*",
    "uuid": "*",
    "yup": "*"
  },
  "scripts": {
    "eas-build-pre-install": "cd ../../ && cp yarn.lock ./apps/mobile-app/",
    "postinstall": "rm -r node_modules && cd ../../ && yarn install && npx nx sync-deps mobile-app && npx nx ensure-symlink mobile-app",
    "start": "expo start --dev-client",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "prebuild": "APP_ENV=production expo prebuild --clean",
    "prebuild:dev": "APP_ENV=development expo prebuild --clean",
    "build:ios": "yarn eas build --platform ios --local --non-interactive --clear-cache",
    "build:android": "yarn eas build --platform android --local --non-interactive --clear-cache",
    "build:ios:dev": "yarn eas build --profile development --platform ios --local --non-interactive --clear-cache",
    "build:android:dev": "yarn eas build --profile development --platform android --local --non-interactive --clear-cache",
    "submit:ios": "yarn eas submit --platform ios",
    "submit:android": "yarn eas submit --platform android"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9"
  }
}

**Comment by atanaskanchev** [comment]
Couldn't get a remote EAS iOS simulator build to work

Build ID
ffb0b4f3-0e70-4f80-a244-6073448deb29

```
Waiting to start
1h 4m 34s

Spin up build environment
3m 24s

Pre-install hook
1s

Read package.json
1s

Install dependencies
7m 38s

Read app config
1s

Run expo doctor
13s

Prebuild
1s

Install pods
15m 59s

Configure expo-updates
1s

Run fastlane
20m 59s

Fail build
1s

Build failed: Something went wrong, marking build as errored
Your build exceeded the maximum build time of 45 minutes.
```

**Comment by izakfilmalter** [comment]
I am still getting this error after updating to `47.0.6`.

**Comment by megan-woodruff** [comment]
I am also still seeing the issue

**Comment by tjdownes** [comment]
I am also getting this with 47.0.6, iOS builds.

**Comment by nicolasmelo1** [comment]
I am algo getting this error

`eas build  --profile qa --platform ios --local` runs fine and build the app on my machine
`eas build  --profile qa --platform ios --clear-cache` fails when i try to run on EAS Cloud.

The error for me on EAS is:
```
error: File /Users/expo/Library/Developer/Xcode/DerivedData/Example/Build/Intermediates.noindex/ArchiveIntermediates/Example-ckxmuakldjajdkjhasddjw/BuildProductsPath/Release-iphoneos/Example.app/main.jsbundle does not exist. This must be a bug with React Native, please report it here: https://github.com/facebook/react-native/issues
```

**Comment by tjdownes** [comment]
I noticed this in the log, unsure if it's related:

```
[RUN_FASTLANE] › Copying   ../../Library/Developer/Xcode/DerivedData/OutdoorAwarenessKit-exukpwimbaodybgrvuvvgtzmubzk/Build/Intermediates.noindex/ArchiveIntermediates/OutdoorAwarenessKit/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/EXDevLauncher.bundle/main.jsbundle ➜ ../../../../var/folders/x1/ywsdnd_x419gb7k5vjxls0340000gn/T/eas-build-local-nodejs/f4b8642a-bd49-4142-84a9-f7a3e7704cf1/build/node_modules/expo-dev-launcher/ios/main.jsbundle
```
are the paths in the above supposed to match the ones in the errors?

```❌  error: File /Users/tjdownes/Library/Developer/Xcode/DerivedData/OutdoorAwarenessKit-exukpwimbaodybgrvuvvgtzmubzk/Build/Intermediates.noindex/ArchiveIntermediates/OutdoorAwarenessKit/BuildProductsPath/Release-iphoneos/OutdoorAwarenessKit.app/main.jsbundle does not exist. This must be a bug with React Native, please report it here: https://github.com/facebook/react-native/issues```

**Comment by izakfilmalter** [comment]
Has anyone figured out a solution for this?

**Comment by tjdownes** [comment]
> Has anyone figured out a solution for this?

I've been digging into it, but no solution thus far. This is my first actual production build, so it's the first time I encountered it. I am working to understand how that bundle is packaged and pathed, which is part of my blocker to coming up with a solution. I can't even say at this point that it's not something in my dependencies.

I did try rolling back to previous SDK and dependencies, but that did not fix it. So in my case this is not specific to SDK 47.0.6

**Comment by lindan4** [comment]
Oddly enough it looks like the PR that addressed this build may have been specific to my configuration.

**Comment by izakfilmalter** [comment]
@lindan4 did the or fix the build error for you? Or is it still broken?

**Comment by fencer-yd** [comment]
is this problem solved

**Comment by tjdownes** [comment]
No, I don't believe it is. I believe The problem still exists

**Comment by lindan4** [comment]
It has been fixed for me

**Comment by izakfilmalter** [comment]
```

**Comment by fencer-yd** [comment]
I checked my xcode log, because the js parsing error was reported. It's working fine now.

**Comment by byCedric** [comment]
This issue was fixed in #20056, and released in `expo@47.0.6`. Please make sure you are using the fixed package version any of these commands:
- `$ npm ls expo`
- `$ pnpm list expo`
- `$ yarn why expo`
- `$ node --print "require('expo/package.json').version"`

If you are still running into similar issues, please open a new issue. I'll lock this issue to keep these reports separate.

Hope this helps!
